src/cpp/fastdds/publisher/DataWriterHistory.hpp

```diff
@@ -87,6 +87,49 @@ class DataWriterHistory : public fastrtps::rtps::WriterHistory
             std::unique_lock<fastrtps::RecursiveTimedMutex>& lock,
             const std::chrono::time_point<std::chrono::steady_clock>& max_blocking_time);
 
+    /**
+     * Add a change comming from the DataWriter.
+     *
+     * @param change             Pointer to the change
+     * @param wparams            Extra writer parameters.
+     * @param pre_commit         Functor receiving a CacheChange_t& to perform actions after the
+     *                           change has been added to the history, but before notifying the RTPS writer.
+     * @param lock               Lock to the history mutex.
+     * @param max_blocking_time  Maximum time point to wait for room on the history.
+     *
+     * @return True if added.
+     */
+    template<typename PreCommitHook>
+    bool add_pub_change_with_commit_hook(
+            fastrtps::rtps::CacheChange_t* change,
+            fastrtps::rtps::WriteParams& wparams,
+            PreCommitHook pre_commit,
```

Comment:
Because converting a lambda into `std::function` may imply dynamic allocations. See [this blog](https://blog.demofox.org/2015/02/25/avoiding-the-performance-hazzards-of-stdfunction/) for some insights.

Reply:
Nice, the more you know!